src/lib/caching/structures/guilds/GuildMember.ts

```diff
@@ -85,6 +85,75 @@ export class GuildMember extends Structure {
 		return this.client.users.get(this.id) ?? null;
 	}
 
+	/**
+	 * The calculated permissions from the member's {@link GuildMemberRoleStore roles}.
+	 * @since 0.0.1
+	 */
+	public get permissions(): Readonly<Permissions> {
+		if (this.id === this.guild.ownerID) return new Permissions(Permissions.ALL).freeze();
+
+		const permissions = new Permissions(this.roles.map(role => role.permissions));
+		return (permissions.has(Permissions.FLAGS.ADMINISTRATOR) ? permissions.add(Permissions.ALL) : permissions).freeze();
+	}
+
+	/**
+	 * Whether or not the {@link ClientUser client user} can manage this member. This is based on:
+	 * - The member is not the {@link Guild#owner guild owner}.
+	 * - The member is not the {@link ClientUser client user} itself.
+	 * - The {@link ClientUser client user} is the owner of the {@link Guild}.
+	 * - The {@link ClientUser client user}'s {@link GuildMemberRoleStore#highest highest role} is higher than the member's.
+	 * @since 0.0.1
+	 * @returns `true` when any of the conditions are met, `null` when the {@link ClientUser client user}'s member is not
+	 * cached (or when {@link Client#user} is null), or `false` otherwise.
+	 */
+	public get manageable(): boolean | null {
+		// If the user is the owner, it is not manageable.
+		if (this.id === this.guild.ownerID) return false;
+
+		// If the client user is not cached, return null.
+		const { user } = this.client;
+		if (!user) return null;
+
+		// If the user is the same as the client user, it is not manageable.
+		if (this.id === user.id) return false;
+
+		// If the client user's member instance is not cached, return null.
+		const { me } = this.guild;
+		if (!me) return null;
```

Comment:
What are the chances of this returning null if `this.id` comes from the user?

Reply:
That the client user is cached, but the member is not.